### DISCUSSION TITLE
Mitigate breaking change on 11.0 version

### DISCUSSION THREAD

**Discussion by deandreamatias** [flutter-form-builder-ecosystem/form_builder_validators#117] [Ideas]
- The `checkNullOrEmpty` breaks a lot (more than expected) the use of package
- With `checkNullOrEmpty: true`, the default behaviour, the use of FormBuilderValidators.required() won't make sense
- Maybe could be mitigated with `dart fix` as described in [Data-driven Fixes](https://github.com/flutter/flutter/blob/master/docs/contributing/Data-driven-Fixes.md).

Related with: #116 

@techouse @martijn00 can continue the discussion here I guess

**Comment by ArturAssisComp** [comment]
I suggested an API change in #128. It removes the necessity of checkNullOrEmpty. What do you think?

**Reply by techouse** [reply]
I like the idea.

**Reply by deandreamatias** [reply]
I think that is a nice way to improve the package

**Reply by ArturAssisComp** [reply]
Thus I will open a draft PR to start implementing a proof of concept for this idea. I will implement it under the hood of the current API to check if it is feasible, making it possible to use all the test cases already implemented. And then, if it works properly, it can be exposed to the final users. What do you think? @deandreamatias @martijn00

**Reply by deandreamatias** [reply]
I wil really appreciate if you do this

**Comment by deandreamatias** [comment]
The #129 big PR will be merge.

The next steps:

1. Decide to mantain bot APIs or remove complety the old
2. Improve documentation readme about migration
3. Try to implement migration with dart fix. I implemented on [flutter_form_builder](https://github.com/flutter-form-builder-ecosystem/flutter_form_builder/blob/main/lib/fix_data.yaml), so I can work on this
4. Remove old API
5. Prepare prerelease, updating changelog with breaking changes and version

What do you think? @ArturAssisComp and everyone else that want participate.
For each of them, we can create new PRs. I can take care of some of them.

**Reply by ArturAssisComp** [reply]
Ok, maybe we can deprecate the old API. In the future it could be removed.
Either way, it will be necessary to do the item 2. So I think we can start that.

**Comment by ArturAssisComp** [comment]
Can I initiate the item 3?

**Reply by deandreamatias** [reply]
Hi @ArturAssisComp if do you want, yes, can initiate

**Reply by deandreamatias** [reply]
I created a new branch for this changes: [dart-fixes](https://github.com/flutter-form-builder-ecosystem/form_builder_validators/tree/dart-fixes)

**Reply by ArturAssisComp** [reply]
Ok, I will start working on that.
